services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/resource_tracking/core.py

```diff
@@ -0,0 +1,129 @@
+import logging
+from typing import Final
+
+from fastapi import FastAPI
+from models_library.generated_models.docker_rest_api import ContainerState
+from models_library.rabbitmq_messages import (
+    RabbitResourceTrackingHeartbeatMessage,
+    RabbitResourceTrackingStartedMessage,
+    RabbitResourceTrackingStoppedMessage,
+    SimcorePlatformStatus,
+)
+from models_library.services import ServiceType
+from models_library.services_creation import CreateServiceMetricsAdditionalParams
+from pydantic import NonNegativeFloat
+from servicelib.background_task import start_periodic_task, stop_periodic_task
+from servicelib.logging_utils import log_context
+
+from ...core.docker_utils import (
+    are_all_containers_in_expected_states,
+    get_container_states,
+)
+from ...core.rabbitmq import post_resource_tracking_message
+from ...core.settings import ApplicationSettings
+from ...models.shared_store import SharedStore
+from ._models import ResourceTrackingState
+from .settings import ResourceTrackingSettings
+
+_STOP_WORKER_TIMEOUT_S: Final[NonNegativeFloat] = 1.0
+
+_logger = logging.getLogger(__name__)
+
+
+def _get_settings(app: FastAPI) -> ApplicationSettings:
+    settings: ApplicationSettings = app.state.settings
+    return settings
+
+
+async def _start_heart_beat_task(app: FastAPI) -> None:
+    settings: ResourceTrackingSettings = app.state.settings.RESOURCE_TRACKING
+    resource_tracking: ResourceTrackingState = app.state.resource_tracking
+
+    if resource_tracking.heart_beat_task is not None:
+        msg = f"Unexpected task={resource_tracking.heart_beat_task} already running!"
+        raise RuntimeError(msg)
+
+    with log_context(_logger, logging.DEBUG, "starting heart beat task"):
+        resource_tracking.heart_beat_task = start_periodic_task(
+            _heart_beat_task,
+            app=app,
+            interval=settings.RESOURCE_TRACKING_HEARTBEAT_INTERVAL,
+            task_name="resource_tracking_heart_beat",
+        )
+
+
+async def stop_heart_beat_task(app: FastAPI) -> None:
```

Comment:
Why having `__stop_heart_beat_task` and `stop_heart_beat_task`? I really do not see the use of this. Why not just the latter?